File: exercism/go/secret-handshake/secret_handshake.go
package secret

const (
	wink = 1 << iota
	doubleBlink
	closeYourEyes
	jump
	reverse
)

func Handshake(code uint) (ret []string) {
	if code&wink != 0 {
		ret = append(ret, "wink")
	}
	if code&doubleBlink != 0 {
		ret = append(ret, "double blink")
	}
	if code&closeYourEyes != 0 {
		ret = append(ret, "close your eyes")
	}
	if code&jump != 0 {
		ret = append(ret, "jump")
	}
	if code&reverse != 0 {
		for i := len(ret)/2 - 1; i >= 0; i-- {
			opp := len(ret) - 1 - i
			ret[i], ret[opp] = ret[opp], ret[i]
		}
	}
	return
}

File: exercism/c/secret-handshake/src/secret_handshake.c
#include "secret_handshake.h"
#include <stdlib.h>

const int MAX_ARR_LEN = 4;
char* actions[] = {"wink", "double blink", "close your eyes", "jump"};

const char** commands(size_t number)
{
    char** ret = malloc(sizeof(char*) * MAX_ARR_LEN);
    ret[0] = NULL;

    int p = 0;
    for (int i = 0; i < 4; ++i) {
        int idx = (number & 1 << 4) ? 4-1-i : i;
        if (number & 1 << idx) {
            ret[p++] = actions[idx];
        }
    }
    return (const char**) ret;
}
